alea/submitters/htcondor.py

```diff
@@ -638,10 +636,11 @@ def _generate_workflow(self, name="run_toymc_wrapper"):
 
             # Add the arguments into the job
             # Using escaped argument to avoid the shell syntax error
-            def _extract_all_to_tuple(d):
+            def _extract_all_to_tuple(kwargs) -> tuple:
+                """Generate the submission script from the runner arguments."""
                 return tuple(
-                    f"{json.dumps(str(d[key])).replace(' ', '')}".replace("'", '\\"')
-                    for key in d.keys()
+                    shlex.quote(Submitter.arg_to_str(kwargs[arg], annotation))
```

Comment:
Similar to https://github.com/XENONnT/alea/blob/09f35f50c9fe61769b20ddd69371ac31876ee307/alea/submitter.py#L579